python/pyarrow/_csv.pyx

```diff
@@ -1288,20 +1301,25 @@ cdef class WriteOptions(_Weakrefable):
         CSV data
     delimiter : 1-character string, optional (default ",")
         The character delimiting individual cells in the CSV data.
+    quoting_style : str, optional (default "needed")
+        Whether to quote values, and if so, which quoting style to use.
+        Accepted values are "needed", "all_valid", "none"
```

Comment:
We would probably like to expand the explanation a bit, for example:
```suggestion
        The following values are accepted:
        - "needed" (default): only enclose values in quotes when needed.
        - "all_valid": enclose all valid values in quotes; nulls are not quoted.
        - "none": do not enclose any values in quotes; values containing
          special characters (such as quotes, cell delimiters or line endings)
          will raise an error.
```

Reply:
Done, thanks for writing this out.